src/test/java/net/kaczmarzyk/spring/data/jpa/utils/SpecificationBuilderTest.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2014-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.kaczmarzyk.spring.data.jpa.utils;
+
+
+import net.kaczmarzyk.E2eTestBase;
+import net.kaczmarzyk.spring.data.jpa.Customer;
+import net.kaczmarzyk.spring.data.jpa.domain.In;
+import net.kaczmarzyk.spring.data.jpa.web.annotation.Join;
+import net.kaczmarzyk.spring.data.jpa.web.annotation.Spec;
+import org.junit.Test;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.util.List;
+
+import static net.kaczmarzyk.spring.data.jpa.utils.SpecificationBuilder.specification;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Kacper Leśniak
+ */
+public class SpecificationBuilderTest extends E2eTestBase {
```

Comment:
does it have to extend E2E test?

Reply:
is it enough reason to extend E2eTestBase? Do you need features from E2eTestBase? Perhaps it's enough to extend IntegrationTestBase and create customers directly in the test?